examples/mistral/user_script.py

```diff
@@ -0,0 +1,165 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import random
+from typing import List, Tuple
+
+import numpy as np
+import torch
+from datasets import load_dataset
+from transformers import AutoConfig, LlamaTokenizer
+
+model_id = "mistralai/Mistral-7B-v0.1"
+config = AutoConfig.from_pretrained(model_id)
+
+
+def calib_dataloader(data_dir, batch_size, *args, **kwargs):
+    model_path = kwargs.pop("model_path")
+    return PileDataloader(model_path, batch_size=batch_size)
+
+
+def create_dataloader(data_dir, batchsize, *args, **kwargs):
+    dataloader = PileDataloader(model_id, batch_size=batchsize, seq_len=32, past_seq_len=32, sub_folder="train")
+    for data, label in dataloader:
+        d = {name: to_numpy(inp_data) for name, inp_data in data.items()}
+        yield d, label
+
+
+def tokenize_function(examples):
+    tokenizer = LlamaTokenizer.from_pretrained(model_id)
+    return tokenizer(examples["text"])
+
+
+class PileDataloader:
+    def __init__(self, model_path, batch_size=1, seq_len=32, past_seq_len=32, sub_folder="train"):
+        random.seed(0)
+        self.batch_size = batch_size
+        self.seq_len = seq_len
+        self.past_seq_len = past_seq_len
+
+        dataset = load_dataset("NeelNanda/pile-10k", split=sub_folder)
+        self.dataset = dataset.map(tokenize_function, batched=True)
+        self.dataset.set_format(type="torch", columns=["input_ids", "attention_mask"])
+
+    def __iter__(self):
+        length = len(self.dataset)
+        counter = 0
+
+        while counter < length:
+            while True:
+                i = random.randint(0, len(self.dataset) - 1)
+                trainenc = self.dataset[i]
+                if trainenc["input_ids"].shape[0] > self.seq_len:
```

Comment:
Perhaps we can add a judgment that all input data does not meet the conditions to avoid an infinite loop in the while True.

Reply:
what's your suggestion to avoid the infinite loop?